lib/Dialect/TTIR/Analysis/LegalGridAnalysis.cpp

```diff
@@ -36,17 +70,95 @@ bool LegalGridAnalysis::applyOverrides() {
 }
 
 void LegalGridAnalysis::analysisImplementation() {
-  // Placeholder, needs to be implemented. Go through all the grid sizes and
-  // check if they are legal based on tensor type and device/chip attributes.
-  // For now result of analysis is maximum supported grid size.
-  //
+  // A first incomplete implementation of the LegalGridAnalysis.
+  // This implementation is a placeholder and is meant to just enable testing of
+  // other components.
+
+  // Skip mlir ops.
+  if (not llvm::isa<TTIROp>(op)) {
+    return;
+  }
+  // Skip operations that don't have output tensors.
```

Comment:
Can this be used instead:
` if (op->getNumResults() == 0)`

Reply:
Yes, except for ToLayoutOp which has an output.